internal/pkg/template/diff/diff.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"gopkg.in/yaml.v3"
+)
+
+// DiffNode represents a difference between two YAML documents.
+type DiffNode struct {
+	key      string
+	children map[string]*DiffNode // A list of non-empty pointers to the children nodes.
+
+	oldValue *yaml.Node // Only populated for a leaf node.
```

Comment:
out of curiosity: how do we define a leaf node?

Reply:
By `len(children) == 0`. I can add this to the field comment.

Reply:
derp that makes sense!